autopilot/contractor.go

```diff
@@ -1140,10 +1138,19 @@ func (c *contractor) formContract(ctx context.Context, w Worker, host hostdb.Hos
 	return formedContract, true, nil
 }
 
-func (c *contractor) priceTable(ctx context.Context, w Worker, hk types.PublicKey, siamuxAddr string) (rhpv3.HostPriceTable, error) {
+func (c *contractor) priceTable(ctx context.Context, w Worker, hk types.PublicKey, siamuxAddr string) (*hostdb.HostPriceTable, error) {
 	ctx, cancel := context.WithTimeout(ctx, contractHostPriceTableTimeout)
 	defer cancel()
-	return w.RHPPriceTable(ctx, hk, siamuxAddr)
+
+	pt, err := w.RHPPriceTable(ctx, hk, siamuxAddr)
+	if err != nil {
+		return nil, err
+	}
+
+	return &hostdb.HostPriceTable{
+		HostPriceTable: pt,
+		Expiry:         time.Now().Add(pt.Validity),
+	}, nil
```

Comment:
I feel an RHP route should return the RHP type and not some `hostdb` type.

Reply:
Not in particular. Just felt like decorating a pricetable with its expiry can be done right when we receive it from the host since it's useful info to have everywhere.

Reply:
I don't mind changing it, it's just something I tripped over that's probably not important. It feels weird not returning the "raw" rhpv3 type on an `RHP` route.